src/harness/testcases/heartbeat_testcase.py

```diff
@@ -400,3 +401,100 @@ def test_10_9_4_2_3_1_3(self):
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    request = {'registrationRequest': [device_a]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+
+    # Request grant
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_id
+    request = {'grantRequest': [grant_0]}
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse'][0]
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['response']['responseCode'], 0)
+    grant_id = response['grantId']
+    grant_expire_time = datetime.strptime(response['grantExpireTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+    del request, response
+
+    # First successful Heartbeat
+    request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': 'GRANTED'
+        }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    # Check the heartbeat response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    self.assertLess(datetime.utcnow(),
+                    datetime.strptime(response['transmitExpireTime'],
+                                      '%Y-%m-%dT%H:%M:%SZ'))
+    self.assertEqual(response['response']['responseCode'], 0)
+
+    # Enter the Authorized state
+    request['heartbeatRequest'][0]['operationState'] = 'AUTHORIZED'
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    # Verify successful response
+    transmit_expire_time = datetime.strptime(response['transmitExpireTime'],
+                                             '%Y-%m-%dT%H:%M:%SZ')
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    self.assertLess(datetime.utcnow(), transmit_expire_time)
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+
+    # Relinquish the grant
+    # Note: The Testcase requires grant to be terminated at this point, but
+    # here the grant is being relinquished. The test case document will be
+    # updated later to use this method
+    request = {
+        'relinquishmentRequest': [{
+            'cbsdId': cbsd_id,
+            'grantId': grant_id
+        }]
+    }
+    response = self._sas.Relinquishment(request)['relinquishmentResponse'][0]
+    # Check the relinquishment response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    self.assertEqual(response['response']['responseCode'], 0)
+
+    # use relinquished grantId in new heartbeat request after transmitExpireTime
```

Comment:
On lines 492/493 I calculate how long I have to wait for transmitExpireTime to elapse, by subtracting the current time from it
On line 494 I call sleep() for the calculated amount of time + 1 second
On lines 495 and 496 I am ensuring that the transmitExpireTime has elapsed and the grantExpireTime is still in the future.